octodns_selectel/__init__.py

```diff
@@ -320,20 +320,18 @@ def create_record(self, zone_name, data):
         return self._request('POST', path, data=data)
 
     def delete_record(self, domain, _type, zone):
-        self.log.debug('Delete record. Domain: %s, Type: %s', domain, _type)
+        self.log.debug('Delete records. Domain: %s, Type: %s', domain, _type)
 
         domain_id = self._domain_list[domain]['id']
         records = self._zone_records.get(f'{domain}.', False)
         if not records:
             path = f'/{domain_id}/records/'
             records = self._request('GET', path)
 
+        full_domain = f'{zone}.{domain}' if zone else domain
         for record in records:
-            full_domain = domain
-            if zone:
-                full_domain = f'{zone}.{domain}'
             if record['type'] == _type and record['name'] == full_domain:
                 path = f'/{domain_id}/records/{record["id"]}'
-                return self._request('DELETE', path)
+                self._request('DELETE', path)
 
-        self.log.debug('Delete record failed (Record not found)')
+        self.log.debug('Delete records failed (Record not found)')
```

Comment:
I believe this line will always be reached now that it's not returning at the first matching delete. I assume it'll need to keep track of whether anything was found or not. Perhaps a counter that counts the number of things deleted and then the message can be updated to something like:

```python
if count == 0:
    self.log.warning('Delete records failed (Record not found)')
else:
    self.log.debug('Deleted %d records', count)
```

or something like that.

Reply:
changed